search-filter.lua

```diff
@@ -0,0 +1,108 @@
+-- Required libraries
+local gui = require('gui') -- Importing the 'gui' library
+local widgets = require('gui.widgets') -- Importing the 'widgets' library
+
+-- Define SearchEngine class
+SearchEngine = defclass(SearchEngine, widgets.Window) -- Defining a new class 'SearchEngine' that inherits from 'widgets.Window'
+SearchEngine.ATTRS = {
+    frame_title='Search Engine for Fort-Controlled Units', -- Title of the frame
+    frame={w=50, h=45}, -- Dimensions of the frame
+    resizable=true, -- Frame can be resized
+    resize_min={w=43, h=20}, -- Minimum dimensions when resizing
+}
+
+function SearchEngine:init() -- Initialization function for the SearchEngine class
+    self:addviews{
+        widgets.EditField{
+            view_id='edit',
+            frame={t=1, l=1}, -- Position of the EditField view
+            text='', -- Initial text in the EditField view
+            on_change=self:callback('updateList'), -- Callback function when text in EditField changes
+        },
+        widgets.List{
+            view_id='list',
+            frame={t=3, b=0},
+            choices=self:getFortControlledUnits(), -- Choices in the List view are obtained from getFortControlledUnits function
+            on_select=self:callback('onSelect'), -- Callback function when a unit is selected from the list
+        }
+    }
+end
+
+function SearchEngine:getFortControlledUnits()
+    local fortControlledUnits = {}
+    for _, unit in ipairs(df.global.world.units.active) do
+        local unitName = dfhack.TranslateName(dfhack.units.getVisibleName(unit))
+        if unitName == "" then
+            -- Use the race name if the unit's name field is empty
+            unitName = dfhack.units.getRaceName(unit)
+        end
+        if dfhack.units.isVisible(unit) and dfhack.units.isActive(unit) then
+            table.insert(fortControlledUnits, {text=unitName, search_normalized=dfhack.toSearchNormalized(unitName), id=unit.id})
```

Comment:
set `search_key` instead of `search_normalized`, and you don't need to normalize the key yourself with `toSearchNormalized`. that is handled by `widgets.FilteredList`